test/test_torch.py

```diff
@@ -3134,6 +3140,14 @@ def test_torch_from_file(self):
                 t2.fill_(rnum)
                 self.assertEqual(t1, t2, atol=0, rtol=0)
 
+            f = tempfile.NamedTemporaryFile(delete=False)
+            assert_with_filename(f.name)
+            os.unlink(f.name)
+            
+            f2 = tempfile.NamedTemporaryFile(delete=False, dir='中文')
```

Comment:
A better fix is to do `del s1` && `del t1` && `del s2` && `del t2` without specifying `delete=False`.

Reply:
Yeah.. I know it's possible. But personally I try to avoid use `del` as long as it's possible. If you're OK with using `del` to manually dealloc the tensor, I'm with you.

Reply:
Well actually it's a bit complicated than that. For example, this does not work:

```python
with tempfile.NamedTemporaryFile() as f:
    size = 10000
    filename = f.name
    s1 = torch.from_file(filename, True, size, dtype=torch.float)
    # RuntimeError: could not open file <C:\Users\username\AppData\Local\Temp\tmphqvae4pw> in read-write mode; error code: <32>
```

Seems that the temp file is deleted prematurely before torch could actually use it.

Adding `delete=False` will fix this. But when you try to delete it, it will fail because the tensor object is still holding the memory, I assume. So I wrapped it in a function to deliberately make `s` deleted when the function returns:

```python
def test():
    with tempfile.NamedTemporaryFile(delete=False) as f:
        size = 10000
        filename = f.name
        s1 = torch.from_file(filename, True, size, dtype=torch.float)
    return filename

os.unlink(test())
```

Additionaly, `del s1` should also work. But `detele=False` is also required:

```python
with tempfile.NamedTemporaryFile(delete=False) as f:
    size = 10000
    filename = f.name
    s1 = torch.from_file(filename, True, size, dtype=torch.float)
    del s1
os.unlink(filename)
```

Reply:
I think we should use a unified solution for dealing with temporary files. Maybe we should do that later. Currently, as long as it is working, I'm okay with it.

Reply:
There is actually a `torch.testing._internal.common_utils.TemporaryFileName` that handles the Win32 mess. But it lacks the extensibility I need to specify `dir`. And I don't want to bring more changes to this PR.